lighthouse-core/computed/metrics/lantern-first-contentful-paint.js

```diff
@@ -150,7 +150,8 @@ class LanternFirstContentfulPaint extends LanternMetric {
     return dependencyGraph.cloneWithRelationships(node => {
       if (node.type === BaseNode.TYPES.NETWORK) {
         // Exclude all nodes that ended after paintTs (except for the main document which we always consider necessary)
-        if (node.endTime > paintTs && !node.isMainDocument()) return false;
+        if ((node.endTime > paintTs || node.startTime > paintTs) &&
```

Comment:
This deserves a comment. The extra check looks redundant without an explanation.
```suggestion
        // endTime is negative if request does not finish, make sure startTime isn't after paintTs in this case.
        if ((node.endTime > paintTs || node.startTime > paintTs) &&
```

Reply:
Sure, i have added the comment as well as a unit test. The test mocks 2 cases both with an unfinished resource and determines whether the resource is added to both the optimistic and pessimistic lcp graphs. If the start time is after the paint timestamp, the resource should not be included, if the start time is before, it should be added. If the test is not what you had in mind, please let me know.